jbpm-services/jbpm-kie-services/src/main/java/org/jbpm/kie/services/impl/AbstractAdvanceRuntimeDataServiceImpl.java

```diff
@@ -113,25 +109,34 @@ public void setEmf(EntityManagerFactory emf) {
                                                                                                            QueryContext queryContext) {
         BiFunction<StringBuilder, StringBuilder, String> mainSQLProducer;
         Optional<QueryParam> param = findQueryParamMode(attributes);
+        DataCollector<org.jbpm.services.api.model.UserTaskInstanceWithPotOwnerDesc> dataCollector;
         if(param.isPresent() && param.get().getObjectValue().equals("HISTORY")) {
             mainSQLProducer = (derivedTables, globalWhere) -> "SELECT DISTINCT task.taskId as id " +
                                                               " FROM AuditTaskImpl task " +
                                                               " INNER JOIN ProcessInstanceLog pil ON pil.processInstanceId = task.processInstanceId \n " +
                                                               derivedTables +
-                                                              " WHERE  pil.processType = :processType " + globalWhere +
-                                                              " ORDER BY task.taskId ASC ";
-            return queryProcessUserTasksByVariables(attributes, processVariables, taskVariables, owners, processType, varPrefix, queryContext, mainSQLProducer, this::collectHistoryUserTaskData);
+                                                              " WHERE  pil.processType = :processType " + globalWhere;
+            dataCollector = this::collectHistoryUserTaskData;
+            setDefaultSorting (queryContext, " task.taskId");
+           
         } else {
             mainSQLProducer = (derivedTables, globalWhere) -> "SELECT DISTINCT task.id " +
                                                               " FROM Task task " +
                                                               " INNER JOIN ProcessInstanceLog pil ON pil.processInstanceId = task.processInstanceId \n " +
                                                               derivedTables +
-                                                              " WHERE  pil.processType = :processType " + globalWhere +
-                                                              " ORDER BY task.id ASC ";
-            return queryProcessUserTasksByVariables(attributes, processVariables, taskVariables, owners, processType, varPrefix, queryContext, mainSQLProducer, this::collectRuntimeUserTaskData);
+                                                              " WHERE  pil.processType = :processType " + globalWhere;
+            dataCollector = this::collectRuntimeUserTaskData;
+            setDefaultSorting (queryContext, "task.id");
+        }
+        
+        return queryProcessUserTasksByVariables(attributes, processVariables, taskVariables, owners, processType, varPrefix, queryContext, mainSQLProducer, dataCollector);
+    }
+    
+    private void setDefaultSorting (QueryContext queryContext, String orderBy) {
+        if (queryContext.getOrderBy() == null) {
```

Comment:
Shouldn't it be distinct instead of equal?
```suggestion
        if (queryContext.getOrderBy() != null) {
```